doc/setup_instructions.md

```diff
@@ -521,13 +521,16 @@ the pre-commit hook section for how to speed things up.
 Pre-commit hook
 ---------------
 
-On Linux you can enable a pre-commit hook to automatically run the
+You can enable a pre-commit hook to automatically run the
 formatting checks before each commit to avoid accidentally committing
-code with formatting issues. To install the hook run the following
-script:
+code with formatting issues.
+
+To install pre-commit use the commands `pip install pre-commit` or `pip3 install pre-commit`,
+if you are using Linux don't forget to add `--user` flag or it won't work.
```

Comment:
Did you test this? I added this remark to the previous pip usage because it just simply refused to work if I "sudo" installed. If that's not the case here this remark is not needed here.

Reply:
pre-commit works via package manager install, so installing with that must be the appropriate way for linux

Reply:
Installing pre-commit with pip will likely result in much newer versions and everyone using the same one, instead of relying on OS packaged version that can be many years out of date.
What I'm asking is if installing it with "sudo pip install pre-commit" will work or not. Most packages work when installed like that, but babel absolutely refuses to work when installed with "sudo pip".

Reply:
Just looked up, on Ubuntu it seems it is available with the latest 2 OS releases, but it's way different versions:
- 2.6.0-1
- 2.8.2-1
Fedora has:
- 2.3.0 (on Fedora 32)
- 2.5.1 (on Fedora 33)
So it's a mess I don't want to deal with (I want absolutely 0 complaints to me about pre-commit not working). So the instructions should say to install it with pip.